### ISSUE TITLE
Prune old HTLCs from database to reduce size of backups?

### ISSUE THREAD

**Issue by whitslack** [ElementsProject/lightning#4635]
### Issue and Steps to Reproduce

I use PostgreSQL's WAL archiving feature to maintain a mostly-up-to-date off-site backup of my PostgreSQL database. (I understand that I may lose funds in a few channels if ever I should need to restore from an asynchronous database backup.) The WAL archiving works great, but I still need to make full backups periodically, and they are getting quite huge:
```
lightning=> select pg_size_pretty(pg_database_size('lightning')) as db_size, pg_size_pretty(pg_table_size('channel_htlcs')) as htlcs_size;
 db_size | htlcs_size 
---------+------------
 3657 MB | 2713 MB
(1 row)
```

* Is there any reason that I need to retain the details of months-old HTLCs?
* Is it safe to delete records from the `channel_htlcs` table?

**Comment by whitslack** [comment]
I'm about to experiment with expunging the records of old, failed HTLCs, as this is quickly getting out of hand:
```
 db_size | htlcs_size 
---------+------------
 5055 MB | 3873 MB
```

**Comment by whitslack** [comment]
Per https://github.com/ElementsProject/lightning/issues/4824#issuecomment-934235242, it is clear that purging failed HTLCs is a **Bad Idea™**. I didn't go through with it but did notice that records in `channel_htlcs` are automatically deleted (due to `ON DELETE CASCADE`) when the referenced records in `channels` are deleted, which seems to occur when a channel is "forgotten." So that's good. It seems that there is no escaping the need to retain all the "toxic waste" of a long-lived channel.

As an alternative to purging records to reduce backup size, I have partitioned my `channel_htlcs` table using the `id` column as the partition key. This allows partitions spanning earlier ID ranges to be less frequently modified, sometimes allowing them to be excluded from my nightly incremental backup. However, channels still get deleted often enough that all of the partitions are touched on most days, so this is not saving me much backup bandwidth. I wish I could tell PostgreSQL to segregate `channel_htlcs` records into separate filenodes by `channel_id`. Anything to avoid having to back up multiple ~2-GiB table partitions for the umpteenth time just because one channel was closed that day. I've been considering dropping the foreign key constraint in the `channel_htlcs` table so that deleting channels won't cause rows to be deleted from it.

I'll close this issue since my original questions are answered.
